src/custom/hooks/useRefetchPriceCallback.ts

```diff
@@ -190,41 +190,56 @@ function _handleQuoteError({
   clearQuote,
   setQuoteError
 }: HandleQuoteErrorParams) {
+  console.log('Handle error')
+  let quoteError: QuoteError
   if (_isValidOperatorError(error)) {
     switch (error.type) {
       case ApiErrorCodes.UnsupportedToken: {
+        quoteError = 'unsupported-token'
+
         // TODO: will change with introduction of data prop in error responses
         const unsupportedTokenAddress = error.description.split(' ')[2]
         console.error(`${error.message}: ${error.description} - disabling.`)
 
-        return addUnsupportedToken({
+        // Add token to unsupported token list
+        addUnsupportedToken({
           chainId: quoteData.chainId,
           address: unsupportedTokenAddress,
           dateAdded: Date.now()
         })
+        break
+      }
+
+      case ApiErrorCodes.FeeExceedsFrom: {
+        quoteError = 'fee-exceeds-sell-amount'
+        break
       }
-      // Fee/Price query returns error
-      // e.g Insufficient Liquidity or Fee exceeds Price
-      case ApiErrorCodes.FeeExceedsFrom:
+
       case ApiErrorCodes.NotFound: {
+        quoteError = 'insufficient-liquidity'
         console.error(`${error.message}: ${error.description}!`)
-        return setQuoteError({
-          ...quoteData,
-          lastCheck: Date.now(),
-          error: error.type
-        })
+        break
       }
+
       default: {
+        quoteError = 'fetch-quote-error'
         // some other operator error occurred, log it
-        console.error(error)
-        // Clear the quote
-        return clearQuote({ chainId: quoteData.chainId, token: quoteData.sellToken })
+        console.error('Error quoting price/fee. Unhandled operator error: ' + error.type, error)
       }
     }
   } else {
+    quoteError = 'fetch-quote-error'
     // non-operator error log it
-    console.error('An unknown error occurred:', error)
+    console.error('Error quoting price/fee: ' + error)
   }
+
+  // Clear the quote, set UI error
+  clearQuote({ chainId: quoteData.chainId, token: quoteData.sellToken })
+  setQuoteError({
+    ...quoteData,
+    lastCheck: Date.now(),
+    error: quoteError
+  })
```

Comment:
We always clear the quote and set the error in case of any error.

However, note that next PR will make `clearQuote` dissapear, because is not a real action that happens. The action is setQuoteError which will ALSO clear the quote on top of updating the lastCheck and setting the error.